big_tests/tests/graphql_domain_SUITE.erl

```diff
@@ -0,0 +1,218 @@
+-module(graphql_domain_SUITE).
+
+ -include_lib("common_test/include/ct.hrl").
+ -include_lib("eunit/include/eunit.hrl").
+ -include_lib("exml/include/exml.hrl").
+
+ -compile([export_all, nowarn_export_all]).
+
+-import(distributed_helper, [mim/0, require_rpc_nodes/1, rpc/4]).
+-import(graphql_helper, [execute/3, get_listener_port/1, get_listener_config/1]).
+
+-define(HOST_TYPE, <<"dummy auth">>).
+
+suite() ->
+    require_rpc_nodes([mim]) ++ escalus:suite().
+
+ all() ->
+     [{group, domain_handler}].
+
+groups() ->
+     [{domain_handler, [sequence], domain_handler()}].
+
+domain_handler() -> 
+    [create_domain,
+     disable_domain,
+     enable_domain,
+     get_domains_by_host_type,
+     get_domain_details,
+     delete_domain,
+     get_domains_after_deletion].
+
+init_per_suite(Config) ->
+    Endpoint = admin,
+    Opts = get_listener_opts(Endpoint),
+    case proplists:is_defined(username, Opts) of
+        true ->
+            case mongoose_helper:is_rdbms_enabled(?HOST_TYPE) of
+                true -> escalus:init_per_suite([{schema_endpoint, Endpoint} | Config]);
+                false -> {skip, require_rdbms}
+            end;
+        false ->
+            ct:fail(<<"Admin credentials are not defined in config">>)
+    end.
+
+end_per_suite(Config) ->
+    escalus:end_per_suite(Config).
+
+ init_per_testcase(CaseName, Config) ->
+     escalus:init_per_testcase(CaseName, Config).
+
+ end_per_testcase(CaseName, Config) ->
+     escalus:end_per_testcase(CaseName, Config).
+
+create_domain(Config) ->
+    DomainName1 = <<"exampleDomain">>,
+    DomainName2 = <<"exampleDomain2">>,
+
+    Vars = #{domain => DomainName1, hostType => ?HOST_TYPE},
+    Result1 = execute_auth(#{query => create_domain_call(), variables => Vars,
+                   operationName => <<"M1">>}, Config),
+    ParsedResult1 = ok_result(<<"domains">>, <<"addDomain">>, Result1),
+    ?assertEqual(#{<<"domain">> => DomainName1,
+                   <<"hostType">> => ?HOST_TYPE,
+                   <<"enabled">> => null}, ParsedResult1),
+
+    Vars2 = #{domain => DomainName2, hostType => ?HOST_TYPE},
+    Result2 = execute_auth(#{query => create_domain_call(), variables => Vars2,
+                   operationName => <<"M1">>}, Config),
+    ParsedResult2 = ok_result(<<"domains">>, <<"addDomain">>, Result2),
+    ?assertEqual(#{<<"domain">> => DomainName2,
+                   <<"hostType">> => ?HOST_TYPE,
+                   <<"enabled">> => null}, ParsedResult2).
+
+disable_domain(Config) ->
+    Vars = #{domain => <<"exampleDomain">>},
+    Result = execute_auth(#{query => disable_domain_call(), variables => Vars,
+                   operationName => <<"M1">>}, Config),
+    ParsedResult = ok_result(<<"domains">>, <<"disableDomain">>, Result),
+    ?assertEqual(#{<<"domain">> => <<"exampleDomain">>, <<"enabled">> => false}, ParsedResult).
```

Comment:
Could you check that the domain is really disabled here e.g. with calling the node with RPC or using `get_domains_by_host_type`?